Sources/NumberKit/Rational.swift

```diff
@@ -126,23 +126,19 @@ public struct Rational<T: IntegerNumber>: RationalNumber, CustomStringConvertibl
     self.denominator = denominator
   }
 
-  /// Creates a rational number from a numerator and a denominator.
+  /// Creates a rational number from a given numerator and denominator, ignoring overflow; it creates an incorrect
+  /// result iff the correct result is not expressible in the given type.
   public init(_ numerator: T, _ denominator: T) {
     precondition(denominator != 0, "rational with zero denominator")
-    let negative = numerator > 0 && denominator < 0 || numerator < 0 && denominator > 0
-    let anum = Swift.abs(numerator)
-      // numerator < 0 ? -numerator : numerator
-    let adenom = denominator < 0 ? -denominator : denominator
-    let div = Rational.gcd(anum, adenom)
-    self.numerator = negative ? -(anum / div) : (anum / div)
-    self.denominator = adenom / div
+    let (result, _) = Rational.rationalWithOverflow(numerator, denominator) // Ignore overflow.
+    (self.numerator, self.denominator) = (result.numerator, result.denominator)
```

Comment:
You can drop this line if you assign the result of rationalWithOverflow directly to self (in the prior line):
self = Rational.rationalWithOverflow(numerator, denominator).0